### ISSUE TITLE
Make the item universe in train-test files consistent

### ISSUE THREAD

**Issue by mdekstrand** [lenskit/lenskit#912]
Right now, the item universe available in the evaluator is not well-defined in an edge case:

> There’s an interesting corner case in cross-validation that we need to do a bit of design around.
> 
> We have split the data into a train set and test set. A particular user has some train items and some test items.
>  
> We need to set a (default) candidate set, as well as determine what constitutes the item universe when writing an item selector using set operations. This is fairly straightforward when all items are in the training set, but when an item has no training ratings, it’s thorny.
> 
> There are a few options for the item universe:
> -   It is the **training items**
> -   It is the **training items** plus the current user’s **test items**, so it contains all items that they have for testing, but does not contain items that are only in other users’ test sets.
> -   It is **all known items**, including items that only appear in other users’ test sets.
>
> Default candidate set has the same set of options.
>
> Whatever we pick, we need to have a reason and document it. Right now we’re doing the training items. If you want more items, you can configure a different item DAO to have all the items you want to consider in the universe (I think).

We decided to use **all known items**.  Therefore, we need to do the following:

- Make the `ItemDAO` return all known items during the recommender build phase
- Make sure that the `allItems` item selector returns this

**Comment by mdekstrand** [comment]
@vaibhav345 needs this.

**Comment by mdekstrand** [comment]
I'm going to leave this open until it is fully fixed with the rest of the #845 work.

**Comment by mdekstrand** [comment]
This is done with the crossfolder now creating an item set.
